app/file-upload/lib/FileUploadBase.js

```diff
@@ -19,6 +19,11 @@ UploadFS.config.defaultStorePermissions = new UploadFS.StorePermissions({
 			return true;
 		}
 
+		// allows inserts from omnichannel visitors
+		if (doc.visitorToken) {
```

Comment:
you should not allow **any** document just because if has a `visitorToken`, it could lead to undesired results..

I think the permission validation was fixed by https://github.com/RocketChat/Rocket.Chat/pull/19468

can you please test again removing this code and getting the latest changes from `develop`?

Reply:
Thanks @sampaiodiego It seems that the latest changes on develop branch have fixed all the above error which I was facing.

Its also worth mentioning that [this](https://github.com/RocketChat/Rocket.Chat/pull/19429/files#r516425035) change is also not required, since the of [this](https://github.com/RocketChat/Rocket.Chat/blob/48520a394415a5cb812969ddbb0160a4fa172896/app/livechat/server/roomAccessValidator.compatibility.js#L21) validator which `canAccessRoom` calls returns `true` if the visitor belongs to the same room.